src/services/data-loader/data-loader.ts

```diff
@@ -129,22 +129,31 @@ const getS3Connection = () => {
   });
 };
 
-const getAvailableManuscriptPaths = async (client: S3Client): Promise<string[]> => new Promise((resolve, reject) => {
-  const objectsRequest = client.send(new ListObjectsCommand({
-    Bucket: config.s3Bucket,
-    Prefix: 'data/',
-  }));
+const getAvailableManuscriptPaths = async (client: S3Client): Promise<string[]> => {
   const manuscriptPaths: string[] = [];
+  let continuationToken: string | undefined;
+
+  do {
+    const objectsRequest = client.send(new ListObjectsV2Command({
+      Bucket: config.s3Bucket,
+      Prefix: 'data/',
+      ContinuationToken: continuationToken,
+    }));
+
+    // eslint-disable-next-line no-await-in-loop
```

Comment:
As per our discussion with Will, this would perhaps be better served as a recursive function